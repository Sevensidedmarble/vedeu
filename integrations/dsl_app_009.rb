#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

TESTCASE = 'dsl_app_009'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    height 10
    log Dir.tmpdir + '/vedeu_views_dsl.log'
    renderers [
                Vedeu::Renderers::Terminal.new(
                  filename: Dir.tmpdir + "/#{TESTCASE}.out",
                  write_file: true),
                # Vedeu::Renderers::JSON.new(filename: Dir.tmpdir + "/#{TESTCASE}.json"),
                # Vedeu::Renderers::HTML.new(filename: Dir.tmpdir + "/#{TESTCASE}.html"),
                # Vedeu::Renderers::Text.new(filename: Dir.tmpdir + "/#{TESTCASE}.txt"),
              ]
    run_once!
    standalone!
  end

  load File.dirname(__FILE__) + '/support/test_interface.rb'

  Vedeu.render do
    view(:test_interface) do
      stream do
        text 'Multiple ',       { foreground: '#00ff00' }
        text '`text` ',         { foreground: '#007fff' }
        text 'keywords within', { foreground: '#00ff00' }
      end
      stream do
        text 'a ',             { foreground: '#00ff00' }
        text '`stream` ',      { foreground: '#007fff' }
        text 'block view are', { foreground: '#00ff00' }
      end
      stream do
        text 'treated as ',       { foreground: '#00ff00' }
        text 'separate streams.', { foreground: '#007fff' }
      end
    end
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

Vedeu.timer('Test') do
  DSLApp.start
end

load File.dirname(__FILE__) + '/test_runner.rb'
TestRunner.result(TESTCASE, __FILE__)
