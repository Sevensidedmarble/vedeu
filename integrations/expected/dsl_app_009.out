[0m[2;2H[39m[49m(0l(B[2;3H(0q(B[2;4H [2;5HT[2;6He[2;7Hs[2;8Ht[2;9H [2;10H(0q(B[2;11H(0q(B[2;12H(0q(B[2;13H(0q(B[2;14H(0q(B[2;15H(0q(B[2;16H(0q(B[2;17H(0q(B[2;18H(0q(B[2;19H(0q(B[2;20H(0q(B[2;21H(0q(B[2;22H(0q(B[2;23H(0q(B[2;24H(0q(B[2;25H(0q(B[2;26H(0q(B[2;27H(0q(B[2;28H(0q(B[2;29H(0q(B[2;30H(0q(B[2;31H(0q(B[2;32H(0k(B[0m[3;2H[39m[49m(0x(B[3;3H[38;5;46m[49mM[3;4Hu[3;5Hl[3;6Ht[3;7Hi[3;8Hp[3;9Hl[3;10He[3;11H [3;12H[38;5;33m[49m`[3;13Ht[3;14He[3;15Hx[3;16Ht[3;17H`[3;18H [3;19H[38;5;46m[49mk[3;20He[3;21Hy[3;22Hw[3;23Ho[3;24Hr[3;25Hd[3;26Hs[3;27H [3;28Hw[3;29Hi[3;30Ht[3;31Hh[3;32H[39m[49m(0x(B[0m[4;2H[39m[49m(0x(B[4;3H[38;5;46m[49ma[4;4H [4;5H[38;5;33m[49m`[4;6Hs[4;7Ht[4;8Hr[4;9He[4;10Ha[4;11Hm[4;12H`[4;13H [4;14H[38;5;46m[49mb[4;15Hl[4;16Ho[4;17Hc[4;18Hk[4;19H [4;20Hv[4;21Hi[4;22He[4;23Hw[4;24H [4;25Ha[4;26Hr[4;27He[4;32H[39m[49m(0x(B[0m[5;2H[39m[49m(0x(B[5;3H[38;5;46m[49mt[5;4Hr[5;5He[5;6Ha[5;7Ht[5;8He[5;9Hd[5;10H [5;11Ha[5;12Hs[5;13H [5;14H[38;5;33m[49ms[5;15He[5;16Hp[5;17Ha[5;18Hr[5;19Ha[5;20Ht[5;21He[5;22H [5;23Hs[5;24Ht[5;25Hr[5;26He[5;27Ha[5;28Hm[5;29Hs[5;30H.[5;32H[39m[49m(0x(B[0m[6;2H[39m[49m(0m(B[6;3H(0q(B[6;4H(0q(B[6;5H(0q(B[6;6H(0q(B[6;7H(0q(B[6;8H(0q(B[6;9H(0q(B[6;10H(0q(B[6;11H(0q(B[6;12H(0q(B[6;13H(0q(B[6;14H(0q(B[6;15H(0q(B[6;16H(0q(B[6;17H(0q(B[6;18H(0q(B[6;19H(0q(B[6;20H(0q(B[6;21H(0q(B[6;22H(0q(B[6;23H(0q(B[6;24H(0q(B[6;25H(0q(B[6;26H(0q(B[6;27H(0q(B[6;28H(0q(B[6;29H(0q(B[6;30H(0q(B[6;31H(0q(B[6;32H(0j(B[0m